llvm/unittests/ADT/SmallSetTest.cpp

```diff
@@ -17,6 +17,66 @@
 
 using namespace llvm;
 
+TEST(SmallSetTest, ConstructorIteratorPair) {
+  auto L = {1, 2, 3, 4, 5};
+  SmallSet<int, 4> S(std::begin(L), std::end(L));
+  for (int Value : L)
+    EXPECT_TRUE(S.contains(Value));
```

Comment:
You can also do `EXPECT_THAT(S, UnorderedElementsAreArray(L));`. This prints nice error messages.
Also elsewhere below.